spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -1323,6 +1323,21 @@ public static class Listener {
 		 */
 		private SchemaType schemaType;
 
+		/**
+		 * Max number of messages for batch.
+		 */
+		private int maxNumMessages = -1;
+
+		/**
+		 * Max number of bytes for batch.
```

Comment:
This could be a candidate for `DataSize`, although the unit is implied by the name currently.

Reply:
https://github.com/spring-projects-experimental/spring-pulsar/issues/119